presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java

```diff
@@ -547,19 +549,23 @@ public void checkCanSetCatalogSessionProperty(TransactionId transactionId, Ident
     }
 
     @Override
-    public void checkCanSelectFromColumns(TransactionId transactionId, Identity identity, AccessControlContext context, QualifiedObjectName tableName, Set<String> columnNames)
+    public void checkCanSelectFromColumns(TransactionId transactionId, Identity identity, AccessControlContext context, QualifiedObjectName tableName, Set<Subfield> columnOrSubfieldNames)
     {
         requireNonNull(identity, "identity is null");
         requireNonNull(tableName, "tableName is null");
-        requireNonNull(columnNames, "columnNames is null");
+        requireNonNull(columnOrSubfieldNames, "columnOrSubfieldNames is null");
 
         authenticationCheck(() -> checkCanAccessCatalog(identity, context, tableName.getCatalogName()));
 
-        authorizationCheck(() -> systemAccessControl.get().checkCanSelectFromColumns(identity, context, toCatalogSchemaTableName(tableName), columnNames));
+        authorizationCheck(() -> systemAccessControl.get().checkCanSelectFromColumns(
+                identity,
+                context,
+                toCatalogSchemaTableName(tableName),
+                columnOrSubfieldNames.stream().map(subfield -> subfield.getRootName()).collect(toImmutableSet())));
```

Comment:
noob question. If we are always going after the `rootName` of the `Subfield `why not do a substring on the dot `.` character (for the first occurrence if any) of the input string instead of first creating the `Subfield` and then going after the `rootName`.
I'm not sure if this logic applies to other tokenizers  e.g. Maps and Arrays which requires tokenizing on brackets "[" , "]" for creating Subfields.

Reply:
@ajaygeorge  I have one more PR to put on top of this which will properly use the SubField